include/micm/process/process_set.hpp

```diff
@@ -314,4 +344,30 @@ namespace micm
     }
   }
 
-}  // namespace micm
\ No newline at end of file
+  #ifdef USE_CUDA
+    template<template<class> class MatrixPolicy>
+    inline void ProcessSet::CudaAddForcingTerms(
+        const MatrixPolicy<double>& rate_constants, 
+        const MatrixPolicy<double>& state_variables, 
+        MatrixPolicy<double>& forcing) {
+        micm::cuda::AddForcingTerms_kernelSetup(
+            rate_constants.AsVector().data(),
+            state_variables.AsVector().data(),
+            forcing.AsVector().data(),
+            rate_constants.size(),
+            rate_constants[0].size(),
+            state_variables[0].size(),
+            number_of_reactants_.data(),
+            number_of_reactants_.size(),
+            reactant_ids_.data(),
+            reactant_ids_.size(),
+            number_of_products_.data(),
+            number_of_products_.size(),
+            product_ids_.data(),
+            product_ids_.size(),
+            yields_.data(),
+            yields_.size());
+        }
+    #endif
```

Comment:
To get access to the data members, we added this function here. It passes the needed data to the cuda kernel. And, it means that it's a one line change to go from CPU to GPU. `set.AddForcingTerms` goes to `set.CudaAddForcingTerms`

Reply:
Thanks @K20shores for your clarification. So shall we put these two code in a CUDA file since they are GPU specific?

Reply:
**CudaAddForcingTerms()** needs to be in this file instead of .CU file because the internal function  call needs to access the data members. If we put this function in .CU file, then we needs to pass the data members to the **CudaAddForcingTerms()** and needs to include process_set header file in .CU file, which would cause compiling issue. Correct me if I am wrong? @mattldawson @K20shores

Reply:
Thanks @qinatan for the clarification.